lib/cc/yaml/nodes/engine.rb

```diff
@@ -4,7 +4,28 @@ module Nodes
       class Engine < Mapping
         map :enabled, to: Scalar[:bool], required: true
         map :checks, to: Checks
-        map :config, to: EngineConfig
+        map :config, to: Mapping
+
+        attr_reader :config
+
+        def visit_key_value(visitor, key, value)
+          if key == "config"
+            raw_value = value.to_ruby
+
+            # backwards compatability for phpcodesniffer
+            if raw_value.is_a?(String)
+              @config = { "file" => raw_value }
+            else
+              @config = raw_value
+            end
+          else
+            super
+          end
+        end
+
+        def reload
+          self.merge!("config" => @config)
```

Comment:
I'm interested in why this is useful and the issue could be solved differently within `#visit_key_value` (by putting `self["config"] = @config` somewhere in there, for example). But if you've already thought through that and come to this conclusion, I trust your judgement.